### DISCUSSION TITLE
memory leak when next build

### DISCUSSION THREAD

**Discussion by nonamich** [vercel/next.js#43699] [Help]
My Next.JS application has 136 pages that are generated in parallel with the "next build" command because of this, a lot of memory is leaking and in turn, can crash my instance.

Is it possible to set a limit on the simultaneous generation of pages (concurrency) with the "next build" command?

![image](https://user-images.githubusercontent.com/21929069/205504998-7e3dd22f-3940-4d8f-a6cc-91520e268a59.png)

**Comment by icyJoseph** [comment]
> My Next.JS application has 136 pages that are generated in parallel with the "next build" command because of this, a lot of memory is leaking and in turn, can crash my instance.

Where's the memory leak here? Perhaps, you mean it uses up all of your memory?
 
> Is it possible to set a limit on the simultaneous generation of pages (concurrency) with the "next build" command?

There's experimental flags to control this, via the `experimental.cpus` in `next.config.js`:

```js
/** @type {import('next').NextConfig} */
const nextConfig = {
  reactStrictMode: true,
  swcMinify: true,
  experimental: {
    cpus: 2, // for example
  },
};

module.exports = nextConfig;
```

It must be said tough that the there's already a sensible default for this value based off the OS, and environment gotchas.

I am curious to know if there's an actual memory leak, such as workers continuing to consume memory after next build is done.
